api-server/server/utils/in-memory-cache.test.js

```diff
@@ -0,0 +1,49 @@
+/* global describe expect it beforeEach */
+import inMemoryCache from './in-memory-cache';
+
+describe('InMemoryCache', () => {
+  const theAnswer = 42;
+  const before = 'before';
+  const after = 'after';
+  const emptyCacheValue = null;
+
+  describe('get', () => {
+    it('returns null for an empty cache', () => {
+      const cache = inMemoryCache();
+      expect(cache.get()).toBe(emptyCacheValue);
+    });
+
+    it('returns an initial value', () => {
+      const cache = inMemoryCache(theAnswer);
+      expect(cache.get()).toBe(theAnswer);
+    });
+  });
+
+  describe('update', () => {
+    it('updates the cached value', () => {
+      const cache = inMemoryCache(before);
+      cache.update(() => after);
+      expect(cache.get()).toBe(after);
+    });
+
+    it('can handle promises correctly', done => {
+      const cache = inMemoryCache(before);
+      cache.update(() => new Promise(resolve => resolve(after)));
```

Comment:
This will not work. If you unwrap everything, it will look something like: 

```js
() => Promise.resolve(after))
  .then(() => {
    expect(cache.get()).toBe(after);
    done();	
  })
  .then(value => cache.set(cacheKey, value))
```

Which is an impossible test to pass

Reply:
`update` for promise returns `maybePromisedValue.then(value => cache.set(cacheKey, value))`. Why is `.then(value => cache.set(cacheKey, value))` last in your snippet?

Reply:
```js

update(fn) {
  const value = fn(); 
  /*
  fn = () => Promise.resolve(after).then(() => {
    expect(cache.get()).toBe(after);
    done();	
  })
  */

  if (isPromiseLike(value)) {
    value.then((value) => cache.set(cacheKey, value))

  /*
  which becomes -> 
  Promise.resolve(after)
    .then(() => {
      expect(cache.get()).toBe(after);
      done();	
    })
    .then((value) => cache.set(cacheKey, value))
  */
  }
}
```

Reply:
Ohhhhhhhhhhhh. Pesky brackets!

Fix pushed